private/optimize.m

```diff
@@ -54,39 +54,52 @@
 optimizationSuccess = false;
 iter = 1;
 while ~optimizationSuccess && iter <= 10
-    
-    x0 = getInitialGuess(problem, iter);
-    dispInfo(problem, iter)
-    
-    tic
-    % Symbolic gradients
-% 	[x,fval,exitflag,output,lambda,grad]  = fmincon(@(x) objFun(x), x0, A,b,Aeq,beq,[],[],@(x) feval(nonlconFileName,x,problem.tolIsotropy, ...
-% 											problem.gMaxConstraint, problem.integralConstraint,problem.targetTensor, problem.tolMaxwell*problem.dt^2, ...
-% 											problem.signs),options);
-    % Analytic gradients                                  
-    [x,fval,exitflag,output,lambda,grad]  = fmincon(@(x) objFun(x), x0, A,b,Aeq,beq,[],[], @(x) nonlconAnalytic(x,problem.tolIsotropy, ...
-											problem.gMaxConstraint, problem.integralConstraint,problem.targetTensor, problem.tolMaxwell*problem.dt^2, ...
-											problem.signs, problem.useMaxNorm),options);
-	
-    optimizationTime = toc;
-    
-    disp(['Optimization took ' num2str(optimizationTime, 3) 's.']);
-    
-    optimizationSuccess = (exitflag > 0);
-    
-    iter = iter +1;
-    
-    if ~problem.redoIfFailed
-        if ~optimizationSuccess
-            disp('Optimization failed but will not be repeated!')
+    try
+        x0 = getInitialGuess(problem, iter);
+        dispInfo(problem, iter)
+        
+        tic
+        % Symbolic gradients
+        % 	[x,fval,exitflag,output,lambda,grad]  = fmincon(@(x) objFun(x), x0, A,b,Aeq,beq,[],[],@(x) feval(nonlconFileName,x,problem.tolIsotropy, ...
+        % 											problem.gMaxConstraint, problem.integralConstraint,problem.targetTensor, problem.tolMaxwell*problem.dt^2, ...
+        % 											problem.signs),options);
+        % Analytic gradients
+        [x,fval,exitflag,output,lambda,grad]  = fmincon(@(x) objFun(x), x0, A,b,Aeq,beq,[],[], @(x) nonlconAnalytic(x,problem.tolIsotropy, ...
+            problem.gMaxConstraint, problem.integralConstraint,problem.targetTensor, problem.tolMaxwell*problem.dt^2, ...
+            problem.signs, problem.useMaxNorm, problem.motionCompensation, problem.dt),options);
+        
+        optimizationTime = toc;
+        
+        disp(['Optimization took ' num2str(optimizationTime, 3) 's.']);
+        
+        optimizationSuccess = (exitflag > 0);
+        
+        iter = iter +1;
+        
+        if ~problem.redoIfFailed
+            if ~optimizationSuccess
+                disp('Optimization failed but will not be repeated!')
+            end
+            break
         end
-        break
+        
+    catch
+        iter = iter +1;
```

Comment:
Maybe alert the user that something went wrong.

Reply:
That would be nice!

I will probably fix this when I implement a time-out break. But that is for the future.

Reply:
Feature request